### ISSUE TITLE
Consider sharing a docker image for this

### ISSUE THREAD

**Issue by armandabric** [rezzza/vaultage#25]
I do not do PHP development anymore, so I can't use this easily. Could you provide a docker container to allow using it without the need of the PHP runtime?

**Comment by stephpy** [comment]
You have to edit vaultage source code ?
Else, just use the .phar https://github.com/rezzza/vaultage#installation

**Comment by armandabric** [comment]
Using the phar require a local PHP with the good extension locally installed. With docker we could avoid this kind of local dependencies.

For now I install PHP from brew and use composer global install feature.

I could give a try on a docker image when the issue #26 will be fixed.

**Comment by shouze** [comment]
@Spy-Seth I've already investigated in this direction, but not that easy in fact as you have to forward your gpg-agent socket to the containairized vaultage.

@stephpy yes just the .phar is ok if you have a php binary available with the right requirements on your own machine. docker is great to bundle a software without the need of any dependencies, it just work™.

**Comment by stephpy** [comment]
You're using a docker container for all of your binary/libraries ?

![](http://www.thewwwsite.com/images/0ears1.jpg)

:D

**Comment by shouze** [comment]
@stephpy yes, already true for package dependency managers, for encryption it's maybe another deal in fact, it depends on how things are done. It's a hot subject on my side ATM, I will tell you how I'll manage that ;)

**Comment by gmorel** [comment]
@shouze Did you manage to get it work through a container ?

**Comment by shouze** [comment]
@gmorel in fact we've moved to https://www.vaultproject.io/ as we wanted security as a service & vault is far more complete (vault unseal then various ACL backends).

**Comment by gmorel** [comment]
Nice
